src/app/components/tree/tree.ts

```diff
@@ -791,6 +793,10 @@ export class Tree implements OnInit,AfterContentInit,OnChanges,OnDestroy,Blockab
                 break;
             }
         });
+
+        for (let key in this.parentTemplates) {
+            this.templateMap[key] = this.parentTemplates[key];
```

Comment:
Hi @mortezabahrami , with this for loop you override all the template properties with the same name, is that correct? so if i specify some template only for the tree it will be overridden?

Reply:
Hi @bozzelliandrea, yes and yes.
But I have another question: currently is it possible to pass templates to the Tree while using the Treeselect?

Reply:
But, It's better to  change it this way:
```suggestion
            if (!['header', 'empty', 'footer', 'loader'].includes(key)) {
                    this.templateMap[key] = this.parentTemplates[key];
            }
```

Reply:
@mortezabahrami Yes i think you can customize the template, using styleClass or style bindings for the tree or add your custom template with projection like this
```
<p-treeSelect [(ngModel)]="selectedValue2" selectionMode="multiple" [options]="nodes" placeholder="Select Items">
    <ng-template pTemplate="value">
        Custom Content
    </ng-template>
</p-treeSelect>
```
but honestly I've never had to customize this component that much.

Reply:
The main problem is, the style binding doesn't work at all. And the template only applies to the treeselect input area.
Look at [this](https://stackblitz.com/edit/primeng-treeselect-demo-2a1e1f?file=src/app/app.component.html).